src/validations/css/stylelint.js

```diff
@@ -0,0 +1,65 @@
+import Validator from '../Validator';
+
+const errorMap = {
+  // CSS Syntax Errors
+  'Unclosed block': (error) => {
+    const token = error.source;
+
+    return {
+      reason: 'missing-closing-curly',
+      payload: {error: token},
+    };
+  },
+
+  // Custom Rules
+  'declaration-block-trailing-semicolon': (error) => {
+    const token = error.css;
+
+    return {
+      reason: 'missing-semicolon',
+      payload: {error: token},
+    };
+  },
+};
+
+class StyleLintValidator extends Validator {
+  constructor(source) {
+    super(source, 'css', errorMap);
+  }
+
+  _getRawErrors() {
+    return System.import('../linters').then(
+      ({stylelint}) => stylelint(
+        this._source
+      ).then(
+        (result) => ({result})
+      ).catch(
+        (syntaxError) => ({syntaxError})
+      )
+    );
+  }
+
+  _keyForError(error) {
+    // CSS Syntax Error or Custom Rule
+    return error.reason || (error.messages[0] && error.messages[0].rule);
+  }
+
+  _locationForError(error) {
+    // CSS Syntax Error
+    if (error.reason) {
```

Comment:
I think it would be worth extracting an `isSyntaxError` method, and using it both here and in `_keyForError`